### PR TITLE
Refactor directed_qnet

### PR DESCRIPTION
**Pull request by camall3n**
Here's a suggestion for how we could consolidate that section

### CODE REVIEW

File: actgen/agents/directed_qnet.py [hunk 1]
```diff
@@ -43,51 +43,39 @@ def directed_update(self, states, actions, delta_update, n_updates, baseline_qne
...
-                # update for normal DQN
-                if self.agent_type == 'dqn':
-                    # get q_target
-                    with torch.no_grad():
+                # get q_target
+                with torch.no_grad():
+                    # update for normal DQN
+                    if self.agent_type == 'dqn':
                         original_q_values = self.forward(s.float())
-                        q_target = original_q_values.clone()
```
Comment (zhouzypaul) [inline actgen/agents/directed_qnet.py:51]: I believe we do need the `clone()` method. 

I remember trying without using clone, and `q_target` is just a name reference to the same tensor. Thus, modifying `q_target` also modifies the `original_q_values`

File: actgen/agents/directed_qnet.py [hunk 2]
```diff
@@ -43,51 +43,39 @@ def directed_update(self, states, actions, delta_update, n_updates, baseline_qne
...
                         q_target[0][a] = float(original_q_values[0][a]) + delta_update
 
-                    # perform updates for q(s, a)
-                    self.train()
-                    for i in range(n_updates):
-                        current_q_vals = self.forward(s.float())  # q(s, a') for all a'
+                        get_current_q_vals = lambda s: self.forward(s.float())
```
Comment (zhouzypaul) [inline actgen/agents/directed_qnet.py:54]: ah nice use of the lambda function!

**Comment by zhouzypaul** [comment]
Thanks for the suggestion, this looks good!

File: actgen/agents/directed_qnet.py (hunk 1 above)
Comment (camall3n) [reply actgen/agents/directed_qnet.py:51]: Ah, okay, good catch

Comment (camall3n) [reply actgen/agents/directed_qnet.py:51]: Oh and even worse, I had made it so `q_target` was never initialized...

**Pull request merged**